pippy/LoadModule.py

```diff
@@ -0,0 +1,119 @@
+import os
+import json
+import gc
+from typing import Optional, Union
+
+import torch
+from torch import nn
+
+def load_checkpoint(
+    model: nn.Module,
+    index_filename: Union[str, os.PathLike],
+    device: torch.device = None,
+    dtype: torch.dtype = None,
+):
+    checkpoint_folder = os.path.split(index_filename)[0]
+    with open(index_filename, "r") as f:
+        index = json.loads(f.read())
+    if "weight_map" in index:
+        index = index["weight_map"]
+    checkpoint_files = sorted(list(set(index.values())))
+    checkpoint_files = [os.path.join(checkpoint_folder, f) for f in checkpoint_files]
+    for checkpoint_file in checkpoint_files:
+        checkpoint = torch.load(checkpoint_file)
+        for param_name, param in checkpoint.items():
+            if param_name in ["word_embeddings.weight", "shared.weight"]:
+                if hasattr(model, "lm_head"):
+                    model.lm_head.weight = torch.nn.Parameter((param.clone()).to(device))
+                if hasattr(model, "encoder_embed_tokens"):
+                    model.encoder_embed_tokens.weight = torch.nn.Parameter((param.clone()).to(device))
+                if hasattr(model, "decoder_embed_tokens"):
+                    model.decoder_embed_tokens.weight = torch.nn.Parameter((param.clone()).to(device))
```

Comment:
nit: do you mind putting a comment here on why it would be desirable to make a clone of the parameter for these set of parameters?